commandapi-core/src/main/java/dev/jorel/commandapi/config/DefaultedConfig.java

```diff
@@ -0,0 +1,64 @@
+package dev.jorel.commandapi.config;
+
+import org.jetbrains.annotations.ApiStatus;
+
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+
+@SuppressWarnings("ClassEscapesDefinedScope")
+@ApiStatus.Internal
+public abstract class DefaultedConfig {
+
+	protected final Map<String, CommentedConfigOption<?>> allOptions = new LinkedHashMap<>();
+	protected final Map<String, CommentedSection> allSections = new LinkedHashMap<>();
+
+	public static final CommentedConfigOption<Boolean> VERBOSE_OUTPUTS = new CommentedConfigOption<>(
+		new String[]{
+			"Verbose outputs (default: false)",
+			"If \"true\", outputs command registration and unregistration logs in the console"
+		}, false
+	);
+
+	public static final CommentedConfigOption<Boolean> SILENT_LOGS = new CommentedConfigOption<>(
+		new String[] {
+			"Silent logs (default: false)",
+			"If \"true\", turns off all logging from the CommandAPI, except for errors."
+		}, false
+	);
+
+	public static final CommentedConfigOption<String> MISSING_EXECUTOR_IMPLEMENTATION = new CommentedConfigOption<>(
+		new String[]{
+			"Missing executor implementation (default: \"This command has no implementations for %s\")",
+			"The message to display to senders when a command has no executor. Available",
+			"parameters are:",
+			"  %s - the executor class (lowercase)",
+			"  %S - the executor class (normal case)"
```

Comment:
`normal case`? I feel like this should be changed to `uppercase`

Reply:
No, it is like this in the config now. I mean, we can talk about changing the wording but this was copied over from the currently existing config.

Reply:
I mean, might as well change it whilst we are at it. That is up to you though

Reply:
I mean, the point of this system is that we could update stuff like this later too. For now, I think I'll leave it as is though.

Reply:
And the true/false inconsistencies?

Reply:
Those definitely need to be resolved before merging. While those won't be a problem for existing configs, new configs will get the wrong defaults which obviously shouldn't happen.

Reply:
As a note, `%S` is not uppercase. It is the name of the class, so something like `CraftPlayer`: https://commandapi.jorel.dev/9.5.2/config.html#messages. Technically this is called PascalCase, though "normal case" sounds fine to me for this user-facing comment.